packages/pieces/extract-text/src/lib/actions/native-text-extraction.ts

```diff
@@ -0,0 +1,123 @@
+import { Property, createAction } from '@activepieces/pieces-framework';
+import { parseOfficeAsync } from 'officeparser';
+import { lookup } from 'mime-types';
+import { getDocument, type PDFDocumentProxy } from 'pdfjs-dist';
+import type { TextItem } from 'pdfjs-dist/types/src/display/api';
+
+export const nativeTextExtraction = createAction({
+  name: 'native_text_extraction',
+  description:
+    'Extract text from many a file types (pdf, docx, pptx, xlsx, odt, odp, ods)',
+  displayName: 'Native Extraction',
+  props: {
+    file: Property.File({
+      displayName: 'File',
+      description: 'The file to extract text from',
+      required: true,
+    }),
+    precisePdfBreakingLines: Property.Checkbox({
+      displayName: 'Precise PDF Breaking Lines',
+      description:
+        'The pdf breaking lines annalisis is expensive but should works for small files',
```

Comment:
`Checking this option will analyze PDF breaking lines more accurately. This might slow down processing for large files, but it works well for smaller ones.'`